app/src/main/java/com/lightricks/feedexercise/FeedApp.kt

```diff
@@ -20,6 +21,6 @@ internal class FeedApp : Application() {
 internal fun Context.appComponent(): AppComponent {
     return when (this) {
         is FeedApp -> appComponent
-        else -> this.applicationContext.appComponent()
+        else -> (applicationContext as FeedApp).appComponent
```

Comment:
The when is not needed here. You can replace all the content of the method by `return (applicationContext as FeedApp).appComponent`